homeassistant/components/tado/config_flow.py

```diff
@@ -159,6 +160,56 @@ async def async_step_import(
             },
         )
 
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        self.config_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reconfigure_confirm()
+
+    async def async_step_reconfigure_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        errors: dict[str, str] = {}
+        assert self.config_entry
+
+        if user_input is not None:
+            user_input[CONF_USERNAME] = self.config_entry.data[CONF_USERNAME]
+            try:
+                await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except PyTado.exceptions.TadoWrongCredentialsException:
+                errors["base"] = "invalid_auth"
+            except NoHomes:
+                errors["base"] = "no_homes"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if "base" not in errors:
```

Comment:
```suggestion
            if not errors:
```
would be slightly easier to read and allow for catching specific errors (an option would be to do `errors[CONF_PASSWORD] = "invalid_auth"`).